From 1f00b9e2aa97b5a317520456bd4d85007477251b Mon Sep 17 00:00:00 2001
From: Sergey Shelukhin <sershe@apache.org>
Date: Thu, 8 Oct 2015 11:13:20 -0700
Subject: [PATCH 0352/1164] CDH-32718 HIVE-11149 : Fix issue with sometimes
 HashMap in PerfLogger.java hangs (WangMeng,
 reviewed by Xuefu Zhang, Sergey Shelukhin)

Change-Id: I4a00c851c8627b3ddb1006066602751b7888b3e0
---
 .../org/apache/hadoop/hive/ql/log/PerfLogger.java  |    2 +-
 .../hadoop/hive/ql/session/SessionState.java       |    8 +-------
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/common/src/java/org/apache/hadoop/hive/ql/log/PerfLogger.java b/common/src/java/org/apache/hadoop/hive/ql/log/PerfLogger.java
index 965b583..9085d36 100644
--- a/common/src/java/org/apache/hadoop/hive/ql/log/PerfLogger.java
+++ b/common/src/java/org/apache/hadoop/hive/ql/log/PerfLogger.java
@@ -85,7 +85,7 @@
   protected static final ThreadLocal<PerfLogger> perfLogger = new ThreadLocal<PerfLogger>();
 
 
-  public PerfLogger() {
+  private PerfLogger() {
     // Use getPerfLogger to get an instance of PerfLogger
   }
 
diff --git a/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java b/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java
index 354753e..be4e55a 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java
@@ -206,8 +206,6 @@
    */
   LineageState ls;
 
-  private PerfLogger perfLogger;
-
   private final String userName;
 
   /**
@@ -1405,12 +1403,8 @@ public static PerfLogger getPerfLogger(boolean resetPerfLogger) {
     SessionState ss = get();
     if (ss == null) {
       return PerfLogger.getPerfLogger(null, resetPerfLogger);
-    } else if (ss.perfLogger != null && !resetPerfLogger) {
-      return ss.perfLogger;
     } else {
-      PerfLogger perfLogger = PerfLogger.getPerfLogger(ss.getConf(), resetPerfLogger);
-      ss.perfLogger = perfLogger;
-      return perfLogger;
+      return PerfLogger.getPerfLogger(ss.getConf(), resetPerfLogger);
     }
   }
 
-- 
1.7.9.5

